dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/io/readers/SampleReader.java

```diff
@@ -0,0 +1,187 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.dbsnpimporter.io.readers;
+
+import org.springframework.batch.item.database.JdbcPagingItemReader;
+import org.springframework.batch.item.database.PagingQueryProvider;
+import org.springframework.batch.item.database.support.SqlPagingQueryProviderFactoryBean;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+import uk.ac.ebi.eva.dbsnpimporter.models.Sample;
+
+import javax.sql.DataSource;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+
+
+/**
+ SET search_path TO dbsnp_chicken_9031;
+
+ -- get one ss_id from an assembly and a batch
+ SELECT
+    subind.subsnp_id
+ FROM
+    subind
+    JOIN batch on subind.batch_id = batch.batch_id
+    JOIN subsnp sub ON subind.subsnp_id = sub.subsnp_id
+    JOIN snpsubsnplink link ON sub.subsnp_id = link.subsnp_id
+    JOIN b150_snpcontigloc loc on loc.snp_id = link.snp_id
+    JOIN b150_contiginfo ctg ON ctg.contig_gi = loc.ctg_id
+ WHERE
+    batch.batch_id = $batch_id
+    AND ctg.group_label LIKE $group_label
+    AND ctg.group_term IN $group_term
+ LIMIT 1;
+
+
+ -- get the individuals from the assembly and batch
+ SELECT
+    batch.handle,
+    batch.batch_id,
+    batch.loc_batch_id,
+    indiv.loc_ind_id_upp as individual_name,
+    population.loc_pop_id as population,
+    indiv.ind_id
+    subind.submitted_ind_id
+    ped.pa_ind_id
+    ped.ma_ind_id
+    ped.sex
+ FROM
+    subind
+    JOIN submittedindividual indiv on indiv.submitted_ind_id = subind.submitted_ind_id
+    JOIN pedigreeindividual ped on indiv.ind_id = ped.ind_id
+    JOIN batch on subind.batch_id = batch.batch_id
+    JOIN population on population.pop_id = indiv.pop_id
+    JOIN subsnp sub ON subind.subsnp_id = sub.subsnp_id
+    JOIN snpsubsnplink link ON sub.subsnp_id = link.subsnp_id
+    JOIN b150_snpcontigloc loc on loc.snp_id = link.snp_id
+    JOIN b150_contiginfo ctg ON ctg.contig_gi = loc.ctg_id
+ WHERE
+    sub.subsnp_id = $ss_id
+    AND batch.batch_id = $batch_id
+    AND ctg.group_label = $group_label
+    AND ctg.group_term = $group_term
+ -- LIMIT 5
+ ;
+ */
+public class SampleReader extends JdbcPagingItemReader<Sample> {
+
+    public SampleReader(String dbsnpRelease, int batch, String assembly, List<String> assemblyTypes, DataSource dataSource, int pageSize)
```

Comment:
does it makes sense receive a list of assembly types and just one assembly?

Reply:
yes, it makes sense as we have one DB per assembly, but might want to aggregate all assembly types inside that DB